File: Recipie management system/signup.py
from flask import Flask, request, render_template
import sqlite3
import os
app = Flask(__name__)

BASE_DIR = os.path.dirname(os.path.abspath(__file__))
DATABASE_PATH = os.path.join(BASE_DIR, 'RMS.db')


def init_db():
    with sqlite3.connect(DATABASE_PATH) as conn:
        cursor = conn.cursor()
        cursor.execute('''
            CREATE TABLE IF NOT EXISTS USERS (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                name TEXT NOT NULL,
                email TEXT NOT NULL UNIQUE,
                gender TEXT NOT NULL,
                password TEXT NOT NULL
            );
        ''')
        conn.commit()
        conn.close()

@app.route('/')
def form():
    return render_template('signup.html')


@app.route('/submit', methods=['POST'])
def submit_form():
    name = request.form['username']
    email = request.form['E-mail']
    gender = request.form['gender']
    password = request.form['pass']

  
    conn = sqlite3.connect('data.db')
    cursor = conn.cursor()
    cursor.execute("INSERT INTO users (name, email, gender, password) VALUES (?, ?, ?, ?)", (name, email, gender, password))
    conn.commit()
    conn.close()

    return f"Thank you, {name}. Your data has been saved!"

if __name__ == '__main__':
    init_db()
    app.run(debug=True)

File: Recipie management system/signup.php
<?php
// signup.php
$servername = "localhost";
$username = "root";  // Change according to your phpMyAdmin settings
$password = "";      // Your phpMyAdmin password
$dbname = "recipie"; // Replace with your database name

// Create connection
$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

// Get the POST data from the request
$name = $_POST['username'];
$email = $_POST['E-mail'];
$user_password = $_POST['pass'];
$gender = $_POST['gender'];

// Check if email already exists
$sql = "SELECT * FROM users WHERE email = '$email'";
$result = $conn->query($sql);

if ($result->num_rows > 0) {
    echo "Email already exists.";
} else {
    // Hash the password before saving to the database

    

    // Insert data into the database
    $sql = "INSERT INTO users (user_name, email, gender, pass) VALUES ('$name', '$email', '$gender', '$user_password')";

    if ($conn->query($sql) === TRUE) {
        echo "Sign up successful!";
    } else {
        echo "Error: " . $sql . "<br>" . $conn->error;
    }
}
$conn->close();
?>
